src/map/script.cpp

```diff
@@ -21738,7 +21761,40 @@ BUILDIN_FUNC(instance_create)
 		}
 	}
 
-	script_pushint(st, instance_create(owner_id, script_getstr(st, 2), mode));
+	const char* instance_name = script_getstr(st, 2);
+	int32 result = instance_create(owner_id, instance_name, mode);
+
+	// Handle error messages based on result code
+	if ( sd != nullptr && result < 0 ) { // If there's an error and we have session data
+		uint16 message_id;
+		int32 active_instance_id = script_instancegetid(st, mode);
+		const char* active_instance_name = script_instancegetname(active_instance_id);
```

Comment:
```suggestion
```
This cannot really work?
Unless the NPC/script, which is just trying to create an instance, is already part of an instance?
It would be better to use instance_name variable (defined above the if) direct?